packages/graph/hash_graph/lib/graph/src/provenance/ontology.rs

```diff
@@ -0,0 +1,179 @@
+use serde::{Deserialize, Serialize, Serializer};
+use type_system::{uri::VersionedUri, DataType, EntityType, LinkType, PropertyType};
+use utoipa::ToSchema;
+
+use crate::provenance::{CreatedById, OwnedById, RemovedById, UpdatedById};
+
+/// The metadata required to uniquely identify an ontology element that has been persisted in the
+/// datastore.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct PersistedOntologyIdentifier {
+    #[schema(value_type = String)]
+    uri: VersionedUri,
+    // TODO: owned_by_id is not required to identify an ontology element
+    //  https://app.asana.com/0/1202805690238892/1203214689883091/f
+    owned_by_id: OwnedById,
+}
+
+impl PersistedOntologyIdentifier {
+    #[must_use]
+    pub const fn new(uri: VersionedUri, owned_by_id: OwnedById) -> Self {
+        Self { uri, owned_by_id }
+    }
+
+    #[must_use]
+    pub const fn uri(&self) -> &VersionedUri {
+        &self.uri
+    }
+
+    #[must_use]
+    pub const fn owned_by_id(&self) -> OwnedById {
+        self.owned_by_id
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct PersistedOntologyMetadata {
```

Comment:
I don't think these should have moved to provenance, let's talk about this offline.

Reply:
This is why I made separate commits for this. I reverted these.